examples/with-react-intl/helper/loadIntlMessages.ts

```diff
@@ -1,35 +1,23 @@
-import fs from 'fs/promises'
-import path from 'path'
+export type MessageConfig = Record<string, string>
 
-type LoadI18nMessagesProps = {
-  locale: string
-  defaultLocale: string
-}
-
-type MessageConfig = { [key: string]: string }
-
-export default async function loadI18nMessages({
-  locale,
-  defaultLocale,
-}: LoadI18nMessagesProps): Promise<MessageConfig> {
+export default async function loadI18nMessages(
+  locale: string,
+  defaultLocale = 'en'
+) {
   // If the default locale is being used we can skip it
   if (locale === defaultLocale) {
     return {}
   }
 
-  if (locale !== defaultLocale) {
-    const languagePath = path.join(
-      process.cwd(),
-      `compiled-lang/${locale}.json`
+  try {
+    return import(`compiled-lang/${locale}.json`, {
```

Comment:
Looks like lint is failing from this dynamic import https://github.com/vercel/next.js/actions/runs/3144969067/jobs/5111706076#step:6:463

Reply:
Oh eslint can be a bummer! ESM Type assertions are used by Node.js >= 17. The pipeline runs on Node.js 16.
Nevertheless Next.js is lax when it comes to JSON imports. So the fix in 0135a730d0c6cd399357941bbf0e68d5c373fe59 works for now.

I guess this issue will bubble up as soon as Next.js will enforce ESM Type assertions. 

[More Info](https://github.com/eslint/eslint/discussions/15305)

Reply:
Sadly the following workarounds do not work:

- eslint ignore comments (do not work because it is a syntax error)
- `@babel/plugin-syntax-import-assertions` is ignored by the global eslint parser

Reply:
What I don't understand is that `next/babel` actually enables this behaviour:

https://github.com/vercel/next.js/blob/7e8eb2880f0d885e0487df8196fe913963796fe7/packages/next/build/babel/preset.ts#L157

And this is used inside the eslint config:

https://github.com/vercel/next.js/blob/7e8eb2880f0d885e0487df8196fe913963796fe7/.eslintrc.json#L18